adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -338,9 +340,29 @@ private AuthenticationResult acquireTokenSilentFlow(final AuthenticationRequest
         }
 
         // If we cannot switch to broker, return the result from local flow.
-        if (mBrokerProxy.canSwitchToBroker(authenticationRequest.getAuthority()) == BrokerProxy.SwitchToBroker.CANNOT_SWITCH_TO_BROKER
+        final BrokerProxy.SwitchToBroker switchToBrokerFlag = mBrokerProxy.canSwitchToBroker(authenticationRequest.getAuthority());
+        if (switchToBrokerFlag == BrokerProxy.SwitchToBroker.CANNOT_SWITCH_TO_BROKER
                 || !mBrokerProxy.verifyUser(authenticationRequest.getLoginHint(), authenticationRequest.getUserId())) {
             return authResult;
+        } else if (switchToBrokerFlag == BrokerProxy.SwitchToBroker.NEED_PERMISSIONS_TO_SWITCH_TO_BROKER) {
+            //For android M and above
+            throw new UsageAuthenticationException(
+                        ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING,
+                        "Broker related permissions are missing for GET_ACCOUNTS");
+        }
+
+        //check if App gives permissions of broker usage
```

Comment:
why we're checking again? Didn't the else above already throw the permission missing error?